scvi/model/_amortizedlda.py

```diff
@@ -0,0 +1,247 @@
+
+logger = logging.getLogger(__name__)
+
+
+class AmortizedLDA(PyroSviTrainMixin, BaseModelClass):
+    """
+    Amortized Latent Dirichlet Allocation [Blei03]_.
+
+    Parameters
+    ----------
+    adata
+        AnnData object that has been registered via :func:`~scvi.data.setup_anndata`.
+    n_topics
+        Number of topics to model.
+    n_hidden
+        Number of nodes in the hidden layer of the encoder.
+    cell_topic_prior
+        Prior of cell topic distribution. If `None`, defaults to `1 / n_topics`.
+    topic_gene_prior
+        Prior of topic gene distribution. If `None`, defaults to `1 / n_topics`.
+
+    Examples
+    --------
+    >>> adata = anndata.read_h5ad(path_to_anndata)
+    >>> scvi.data.setup_anndata(adata)
+    >>> model = scvi.model.AmortizedLDA(adata)
+    >>> model.train()
+    >>> gene_by_topic = model.get_gene_by_topic()
+    >>> adata.obsm["X_LDA"] = model.get_latent_representation()
+    """
+
+    def __init__(
+        self,
+        adata: AnnData,
+        n_topics: int = 20,
+        n_hidden: int = 128,
+        cell_topic_prior: Optional[Union[float, Sequence[float]]] = None,
+        topic_gene_prior: Optional[Union[float, Sequence[float]]] = None,
+    ):
+        # in case any other model was created before that shares the same parameter names.
+        pyro.clear_param_store()
+
+        super().__init__(adata)
+
+        n_input = self.summary_stats["n_vars"]
+
+        if (
+            cell_topic_prior is not None
+            and not isinstance(cell_topic_prior, float)
+            and (
+                not isinstance(cell_topic_prior, collections.abc.Sequence)
+                or len(cell_topic_prior) != n_topics
+            )
+        ):
+            raise ValueError(
+                f"cell_topic_prior, {cell_topic_prior}, must be None, "
+                f"a float or a Sequence of length n_topics."
+            )
+        if (
+            topic_gene_prior is not None
+            and not isinstance(topic_gene_prior, float)
+            and (
+                not isinstance(topic_gene_prior, collections.abc.Sequence)
+                or len(topic_gene_prior) != n_input
+            )
+        ):
+            raise ValueError(
+                f"topic_gene_prior, {topic_gene_prior}, must be None, "
+                f"a float or a Sequence of length n_input."
+            )
+
+        self.module = AmortizedLDAPyroModule(
+            n_input=n_input,
+            n_topics=n_topics,
+            n_hidden=n_hidden,
+            cell_topic_prior=cell_topic_prior,
+            topic_gene_prior=topic_gene_prior,
+        )
+
+        self.init_params_ = self._get_init_params(locals())
+
+    def _check_if_not_trained(self):
+        """Check if the model is not trained. Raises a ValueError if not."""
+        if not self.is_trained_:
+            raise ValueError(
+                "Trying to query inferred values from an untrained model. Please train the model first."
+            )
```

Comment:
should we add this to our base model class?

Reply:
Sure